---
#0:OK, others:failed
int32 ret
